tensorbay/opendataset/VGGFace2/loader.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+
+"""Dataloaders of VGGFace2 dataset."""
+
+import csv
+import os
+from collections import defaultdict
+from itertools import islice
+from typing import DefaultDict, Dict, List, Tuple, Generator
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.label import CategoryInfo, Classification, LabeledBox2D, LabeledKeypoints2D
+from tensorbay.label.basic import AttributeType
+from tensorbay.opendataset._utility import glob
+from tensorbay.utility import NameList, chunked
+
+_DATASET_NAME = "VGGFace2"
+_SEGMENTS = ("train", "test")
+
+
+_BOX2D_FILES = (
+    os.path.join("label", "loose_bb_test.csv"),
+    os.path.join("label", "loose_bb_train.csv"),
+)
+_KEYPOINTS2D_FILES = (
+    os.path.join("label", "loose_landmark_test.csv"),
+    os.path.join("label", "loose_landmark_train.csv"),
+)
+
+
+def VGGFace2(path: str) -> Dataset:
+    """`Visual Geometry Group Face 2 <http://www.robots.ox.ac.uk/~vgg/data/vgg_face/>`_ dataset.
+
+    The file structure should be like::
+
+        <path>
+            test_list.txt
+            train_list.txt
+            label/
+                identity_meta.csv
+                loose_bb_test.csv
+                loose_bb_train.csv
+                loose_landmark_test.csv
+                loose_landmark_train.csv
+                attributes/
+                    01-Male.txt
+                    02-Black_Hair.txt
+                    ...
+            test/
+                n000001/
+                    0001_01.jpg
+                    0002_01.jpg
+                    ...
+                n000009/
+                ...
+            train/
+                n000002/
+                    0001_01.jpg
+                    ...
+                n000003/
+                ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(_DATASET_NAME)
+
+    class_id_name, class_name_attrs = _get_classification_info(
+        root_path, (str(item) for item in dataset.catalog.classification.attributes)
+    )
+    # class_id_name: {"n000001:"14th_Dalai_Lama",...}
+    # class_name_attrs: {"14th_Dalai_Lama":{"Sample_Num": 424, "Flag": 0, "Gender": 'm'],...}
+
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    dataset.catalog.classification.categories = NameList(
```

Comment:
we should use `dataset.catalog.classification.add_category` here. I have commented it before, but I think you ignored it.